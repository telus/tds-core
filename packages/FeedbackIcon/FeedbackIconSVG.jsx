import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../shared/utils/safeRest'

const FeedbackIconSVG = ({ a11yText, children, ...rest }) => (
  <svg {...safeRest(rest)} role="img" aria-hidden={a11yText ? undefined : 'true'}>
    <React.Fragment>
      {a11yText && <title>{a11yText}</title>}
      {children}
    </React.Fragment>
  </svg>
)

FeedbackIconSVG.propTypes = {
  a11yText: PropTypes.string,
  children: PropTypes.node.isRequired,
}

FeedbackIconSVG.defaultProps = {
  a11yText: undefined,
}

export default FeedbackIconSVG
