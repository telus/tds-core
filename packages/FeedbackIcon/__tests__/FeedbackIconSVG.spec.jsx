import React from 'react'
import { render, shallow } from 'enzyme'

import FeedbackIconSVG from '../FeedbackIconSVG'

describe('FeedbackIconSVG', () => {
  const doShallow = (props = {}) => shallow(<FeedbackIconSVG {...props} />)

  it('renders', () => {
    const feedbackIconSVG = render(
      <FeedbackIconSVG>
        <path />
      </FeedbackIconSVG>
    )

    expect(feedbackIconSVG).toMatchSnapshot()
  })

  it('to render an svg tag', () => {
    const feedbackIconSVG = doShallow({
      children: <path />,
      id: 'the-id',
      'data-some-attr': 'some value',
    })

    expect(feedbackIconSVG.type()).toEqual('svg')
  })

  it('has an img="role" attribute', () => {
    const feedbackIconSVG = doShallow({
      children: <path />,
    })

    expect(feedbackIconSVG.prop('role')).toEqual('img')
  })

  it('has an aria-hidden="true" attribute if no a11yText is passed', () => {
    const feedbackIconSVG = doShallow({
      children: <path />,
    })

    expect(feedbackIconSVG.prop('aria-hidden')).toEqual('true')
  })

  it('does not have an aria-hidden attribute if a11yText is passed', () => {
    const feedbackIconSVG = doShallow({
      children: <path />,
      a11yText: 'Add Location',
    })

    expect(feedbackIconSVG.prop('aria-hidden')).toBeUndefined()
  })

  it('contains a title tag with a11yText if a11yText is passed', () => {
    const feedbackIconSVG = doShallow({
      children: <path />,
      a11yText: 'Add Location',
    })

    expect(feedbackIconSVG.contains(<title>Add Location</title>)).toEqual(true)
  })

  it('passes additional attributes to the element', () => {
    const feedbackIconSVG = doShallow({
      children: <path />,
      id: 'the-id',
      'data-some-attr': 'some value',
    })

    expect(feedbackIconSVG).toHaveProp('id', 'the-id')
    expect(feedbackIconSVG).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const feedbackIconSVG = doShallow({
      children: <path />,
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(feedbackIconSVG).not.toHaveProp('className', 'my-custom-class')
    expect(feedbackIconSVG).not.toHaveProp('style')
  })
})
