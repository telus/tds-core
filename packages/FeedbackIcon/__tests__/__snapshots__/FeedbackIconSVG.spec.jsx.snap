// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackIconSVG renders 1`] = `
<svg
  aria-hidden="true"
  role="img"
>
  <path />
</svg>
`;
